lib/flight_scheduler/job_registry.rb

```diff
@@ -49,12 +50,13 @@ def add_job(job_id, job)
       if @jobs[job_id]
         raise DuplicateJob, job_id
       end
-      @jobs[job_id] = { job: job, runners: Concurrent::Hash.new }
+      @jobs[job_id] = { job: job, runners: Concurrent::Hash.new, deallocated: false }
     end
 
     def add_runner(job_id, runner_id, runner)
       data = @jobs[job_id]
       raise UnknownJob, job_id if data.nil?
+      raise FrozenJob, job_id if data[:deallocated]
```

Comment:
Should this be `DeallocatedJob`?

Reply:
Good catch: https://github.com/openflighthpc/flight-scheduler-daemon/pull/16/commits/22ad21554ae1181a15c9e10839fc189d1f3c0526